src/sign.rs

```diff
@@ -0,0 +1,25 @@
+use std::time::Duration;
+
+/// On some cases like https://github.com/apache/iceberg-rust/issues/506, user may want to
+/// implement their own signing logic for their own services. This trait is defined to
+/// allow user to inject their own signing logic.
+#[async_trait::async_trait]
+pub trait Sign: 'static {
+    /// Credential type for this signer.
+    type Credential;
```

Comment:
Credential should be `Debug`, `Send`, `Sync`, `Unpin` and `'static`

Reply:
is `Debug` a must?

i found that the `Credential` in aws is marked with `[cfg_attr(test, derive(Debug))]`, and it contains `secret_access_key` `session_token` which is considered confidential. is `[cfg_attr(test, derive(Debug))]` a intended privacy protection over it?

Reply:
Makes sense, let's remove debug from the trait.